### PR TITLE
fix installation package for MacOS

### PR DESCRIPTION
**Pull request by mirogta**
The path was wrong for MacOS, so I've split Linux/MacOS into two separate headings with the correct paths.

### CODE REVIEW

**Review by wallrj — CHANGES_REQUESTED** [review]
Thanks @mirogta 

The problem is that this won't work for MacOS on arm:
 * https://github.com/jetstack/cert-manager/releases/download/v1.6.1/kubectl-cert_manager-darwin-arm64.tar.gz

There are some alternative installation instructions for the `cmctl` command which should work on any platform and architecture.

https://github.com/cert-manager/website/blob/6aa447470a820fd2003230e8bff64428c2cf3a22/content/en/docs/usage/cmctl.md?plain=1#L15-L29


Let's duplicate those here (if they work).

**Comment by mirogta** [comment]
@wallrj thanks - done

I've tested that the downloads work OK.

I was just aware that I personally found it more user friendly to explicitly state which exact OS/arch combination is supported and which isn't. I've added this there as well. Is that better now?

**Comment by mirogta** [comment]
@wallrj build failed because of a spellcheck - can you please advise how to fix it? Or should I just remove the bullet points to get it pass?

```
+++ Running spell check
    content/en/docs/usage/kubectl-plugin.md
       32 | - linux/amd64 
       32 | - linux/amd64 
       33 | - linux/arm64 
       33 | - linux/arm64 
       34 | - darwin/amd64 
       34 | - darwin/amd64 
       35 | - darwin/arm64 
       35 | - darwin/arm64 

>> 8 spelling errors found in 1412 files
```

File: content/en/docs/usage/kubectl-plugin.md [hunk 1]
```diff
@@ -13,24 +13,36 @@ While the kubectl plugin is supported, it is recommended to use
...
+tar xzf kubectl-cert-manager.tar.gz
+sudo mv kubectl-cert_manager /usr/local/bin
 ```
 
+Supported OS and CPU architectures:
+
+- linux/amd64
+- linux/arm64
+- darwin/amd64
+- darwin/arm64
```
Comment (wallrj) [inline content/en/docs/usage/kubectl-plugin.md:35]:
```suggestion
- `linux/amd64`
- `linux/arm64`
- `darwin/amd64`
- `darwin/arm64`
```
This should make the spellchecker happy BUT since we also have `windows/amd64` binaries I think this list might cause confusion. I suggest you drop the list.


Thanks and sorry for the confusing output in CI. See:
 * #802

**Review by wallrj — CHANGES_REQUESTED** [review]

File: content/en/docs/usage/kubectl-plugin.md (hunk 1 above)
Comment (mirogta) [reply content/en/docs/usage/kubectl-plugin.md]: OK done, thanks

File: content/en/docs/usage/kubectl-plugin.md [hunk 2]
```diff
@@ -13,24 +13,29 @@ While the kubectl plugin is supported, it is recommended to use
...
-### macOS/Linux
+### Linux/MacOS
+
 Run the following commands to set up the plugin:
+
 ```console
-$ curl -L -o kubectl-cert-manager.tar.gz https://github.com/jetstack/cert-manager/releases/latest/download/kubectl-cert_manager-linux-amd64.tar.gz
-$ tar xzf kubectl-cert-manager.tar.gz
-$ sudo mv kubectl-cert_manager /usr/local/bin
+OS=$(go env GOOS); ARCH=$(go env GOARCH); curl -L -o kubectl-cert-manager.tar.gz https://github.com/jetstack/cert-manager/releases/latest/download/cmctl-$OS-$ARCH.tar.gz
```
Comment (wallrj) [inline content/en/docs/usage/kubectl-plugin.md:25]:
```suggestion
OS=$(go env GOOS); ARCH=$(go env GOARCH); curl -L -o kubectl-cert-manager.tar.gz https://github.com/jetstack/cert-manager/releases/latest/download/kubectl-cert_manager-$OS-$ARCH.tar.gz
```

Oops. I missed this in my last review. The download URL needs to point at the `kubectl-cert-manager.tar.gz` archive.

```console
$ OS=$(go env GOOS); ARCH=$(go env GOARCH); curl -L -o kubectl-cert-manager.tar.gz https://github.com/jetstack/cert-manager/releases/latest/download/kubectl-cert_manager-$OS-$ARCH.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   171  100   171    0     0    485      0 --:--:-- --:--:-- --:--:--   485
100   676  100   676    0     0   1258      0 --:--:-- --:--:-- --:--:--  660k
100 14.0M  100 14.0M    0     0  7424k      0  0:00:01  0:00:01 --:--:-- 15.2M

```

```console
$ tar --list --file kubectl-cert-manager.tar.gz
LICENSES
kubectl-cert_manager

```

**Review by wallrj — CHANGES_REQUESTED** [review]

File: content/en/docs/usage/kubectl-plugin.md [hunk 3]
```diff
@@ -13,24 +13,29 @@ While the kubectl plugin is supported, it is recommended to use
 [cmctl](../cmctl/) as this enables a better experience via tab auto-completion.
 
 ## Installation
+
 You need the `kubectl-cert-manager.tar.gz` file for the platform you're using, these can be found on our [GitHub releases page](https://github.com/jetstack/cert-manager/releases).
 In order to use the kubectl plugin you need its binary to be accessible under the name `kubectl-cert_manager` in your `$PATH`.
 
-### macOS/Linux
+### Linux/MacOS
```
Comment (wallrj) [inline content/en/docs/usage/kubectl-plugin.md:20]: This will  change the anchors links in the docs and cause broken links for external sites that link directly to this documentation:
 * /docs/usage/kubectl-plugin/#macos-linux VS
 * /docs/usage/kubectl-plugin/#linux-macos

Please revert this change and then I think this can be merged.

File: content/en/docs/usage/kubectl-plugin.md [hunk 4]
```diff
@@ -13,24 +13,29 @@ While the kubectl plugin is supported, it is recommended to use
...
+
 Run the following commands to set up the plugin:
+
 ```console
-$ curl -L -o kubectl-cert-manager.tar.gz https://github.com/jetstack/cert-manager/releases/latest/download/kubectl-cert_manager-linux-amd64.tar.gz
-$ tar xzf kubectl-cert-manager.tar.gz
-$ sudo mv kubectl-cert_manager /usr/local/bin
+OS=$(go env GOOS); ARCH=$(go env GOARCH); curl -L -o kubectl-cert-manager.tar.gz https://github.com/jetstack/cert-manager/releases/latest/download/kubectl-cert_manager-$OS-$ARCH.tar.gz
+tar xzf kubectl-cert-manager.tar.gz
+sudo mv kubectl-cert_manager /usr/local/bin
```
Comment (wallrj) [inline content/en/docs/usage/kubectl-plugin.md:27]: Tested locally and it works. Thanks.

```
OS=$(go env GOOS); ARCH=$(go env GOARCH); curl -L -o kubectl-cert-manager.tar.gz https://github.com/jetstack/cert-manager/releases/latest/download/kubectl-cert_manager-$OS-$ARCH.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   171  100   171    0     0    348      0 --:--:-- --:--:-- --:--:--   348
100   676  100   676    0     0    965      0 --:--:-- --:--:-- --:--:--  660k
100 14.3M  100 14.3M    0     0  3425k      0  0:00:04  0:00:04 --:--:-- 4489k

~/.local/bin/kubectl-cert_manager version --client --short
Client Version: v1.7.0

```

**Review by wallrj — CHANGES_REQUESTED** [review]

File: content/en/docs/usage/kubectl-plugin.md (hunk 3 above)
Comment (mirogta) [reply content/en/docs/usage/kubectl-plugin.md]: Done

**Review by wallrj — APPROVED** [review]
/lgtm
/approve

**Comment by mirogta** [comment]
@def1nity I'm sorry to hear that - the docs were brought in line with the rest of the documentation, namely https://github.com/cert-manager/website/blob/master/content/en/docs/usage/cmctl.md, which also uses `go` (and `curl`).

In the old version, you had to install `curl` on Ubuntu and in the new version you would have to install `curl` and `golang`. Maybe that is an overkill to just get the correct `OS` and `ARCH`. What do you propose?

@wallrj How would you recommend to proceed? Update both the `kubectl-cert-manager` and `cmctl` to only use `curl` without the dependency on `go env`? Or separate the guide for _Linux_ and _MacOS_ (and then possibly even for _Linux/amd64_ and _Linux/arm64_ and _MacOS/amd64_ and _MacOS/arm64_, but as discussed above this would break some reference links to the existing page headers…)

**Comment by def1nity** [comment]
@mirogta Apologies, I'm pretty new to this whole dev, git scenery so I could only share my initial impressions. I've been fighting with linux, k3s, docker, kubectl, awx, networking, proxies, LBs, ingresses and such for a few days now. Loving the whole situation with 95% of all articles/blogs/guides/documents on basic topics being outdated and/or broken/misleading in one way or another... Anyway, I somehow got to the point where certs/https was the next big thing to tackle and while trying to figure out why I couldn't get the plugin to install I stumbled across this #807 by chance as I was following the instructions over at https://cert-manager.io/docs/usage/cmctl/#installation. I didn't encounter any other place or task until now that required or even mentioned golang, even the official looking website I came here from didn't say anything about it. Since I am fairly new, clueless and plenty frustrated at this point I had no idea it was even a thing... I completely disregarded it and now I see that the extension missing from my vm was probably why the cmd was failing... Sorry...

I understand your point about wanting to unify the documentation across common tools and it is totally legit. More info/warnings about such changes in visible places/official channels would improve the user experience a bit. Honestly in the eyes of a noob it just feels like instead of moving on with the deployment I have to stop and figure out why a plugin takes more time to work through than the actual tool :) This got a bit long so I will end with my suggestion - Either place such new requirements like "go" in more and visible places so plebeians like myself can find it easier or just keep both options as long as the old one is viable/working fine and leave the choice to the users. Thanks for asking btw, it was a nice and unexpected change of pace and I'm sorry for the long reply and if I somehow offended you! All the best!

**Comment by wallrj** [comment]
I stumbled on this https://twitter.com/ChrisShort/status/1486436331169501190 and this https://crossplane.io/docs/v1.6/getting-started/install-configure.html#install-crossplane-cli 

> Looking at [@crossplane_io](https://twitter.com/crossplane_io)
>  today. I see something I despise, curl | sh, I pull up the script expecting horrors...
> 
> It's actually very clean and makes sure you get the right version for your environment and nothing more. Less than a full screen of shell. I already like this tool.

I share those feelings. Perhaps we should copy the crossplane install script

* https://github.com/crossplane/crossplane/pull/1813
* https://github.com/crossplane/crossplane/blame/release-1.6/install.sh

**Comment by mirogta** [comment]
@def1nity no worries - there is definitely a lot of potential for improvement! I had my own difficulties with the original documentation, because I've tried to follow it and it didn't work too. Had no idea that the tiny fix would end up raising more issues.

I've raised a [separate issue](https://github.com/cert-manager/website/issues/841) to fix the install and the docs properly, but in the meantime if you just want to install the kubectl plugin (again), you can use this [gist](https://gist.github.com/mirogta/8e79f33f6c6fa9ceafcafd1fe7e10540):
```
url="https://gist.githubusercontent.com/mirogta/8e79f33f6c6fa9ceafcafd1fe7e10540/raw/de023115eede47dd1f31bac92cf771db36a26b80/kubectl-cert-manager-install.sh"
curl -sL $url | sh
sudo mv kubectl-cert_manager /usr/local/bin
kubectl cert-manager --help
```
